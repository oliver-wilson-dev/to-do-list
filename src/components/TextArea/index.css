.textArea {
    resize: none;
    overflow: auto;
    background-color: var(--primaryBackgroundColour);
    font-size: inherit;
    padding: 5px 10px;
    border: 3px solid var(--primaryTextColor);
    margin-right: 5px;
}