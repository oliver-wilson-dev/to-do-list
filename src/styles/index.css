/* This is where any shared styles should live */
@import "./variables.css";

body {
  margin: 0;
  padding: 0;
  background-color: var(--primaryBackgroundColour);
}

h1 {
  font-family: "futura-pt", sans-serif;
}

h2,
h3 {
  font-family: "futura-pt", sans-serif;
  font-weight: 900;
  letter-spacing: 1px;
}

p {
  line-height: 2em;
  font-size: 16px;
}

a {
  text-decoration: none;
  font-family: "futura-pt", sans-serif;
  color: rgb(21, 64, 204);
}