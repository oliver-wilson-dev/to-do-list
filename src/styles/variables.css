:root {
    --primaryBackgroundColour:  #f8f8f8;
    --primaryBorderColour: #0b0a0a;
    --primaryTextColor: #333;
}