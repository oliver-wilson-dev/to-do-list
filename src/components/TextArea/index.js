import React from 'react';
import PropTypes from 'prop-types';
import styles from './index.css';

const TextArea = ({ onChange, value, additionalStyles }) => (
  <textarea
    className={`${styles.textArea} ${additionalStyles || ''}`}
    placeholder="Start typing..."
    maxLength={50}
    onChange={onChange}
    value={value}
  />
);

TextArea.propTypes = {
  additionalStyles: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string
};

export default TextArea;
