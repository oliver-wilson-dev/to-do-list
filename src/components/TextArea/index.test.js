import React from 'react';
import { shallow } from 'enzyme';
import TextArea from './index';
import styles from './index.css';

const mockOnChange = jest.fn();
const mockValue = 'test-value';

const defaultProps = { onChange: mockOnChange, value: mockValue };

const render = overrideProps => shallow(<TextArea {...defaultProps} {...overrideProps} />);

describe('<TextArea/> component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  describe('when there are additional styles provided', () => {
    it('should append them to the className', () => {
      const mockAdditionalClassName = 'test-class';
      expect(render({ additionalStyles: mockAdditionalClassName }).prop('className')).toEqual(`${styles.textArea} ${mockAdditionalClassName}`);
    });
  });
});
