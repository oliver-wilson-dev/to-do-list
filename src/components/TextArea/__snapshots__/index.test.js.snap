// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea/> component should render correctly 1`] = `
<textarea
  className="textArea "
  maxLength={50}
  onChange={[MockFunction]}
  placeholder="Start typing..."
  value="test-value"
/>
`;
